Source/Core/WireframeIndexGenerator.js

```diff
@@ -0,0 +1,234 @@
+  wireframeIndices[index++] = 1;
+
+  // Add two edges for every triangle in the strip
+  for (let i = 0; i < numberOfTriangles; i++) {
+    wireframeIndices[index++] = i + 1;
+    wireframeIndices[index++] = i + 2;
+    wireframeIndices[index++] = i + 2;
+    wireframeIndices[index++] = i;
+  }
+
+  return wireframeIndices;
+}
+
+function createWireframeFromTriangleStripIndices(vertexCount, originalIndices) {
+  const originalIndicesCount = originalIndices.length;
+  const numberOfTriangles = originalIndicesCount - 2;
+  const wireframeIndicesCount = 2 + numberOfTriangles * 4;
+  const wireframeIndices = IndexDatatype.createTypedArray(
+    vertexCount,
+    wireframeIndicesCount
+  );
+  let index = 0;
+
+  // Handle the first edge
+  wireframeIndices[index++] = originalIndices[0];
+  wireframeIndices[index++] = originalIndices[1];
+
+  // Add two edges for every triangle in the strip
+  for (let i = 0; i < numberOfTriangles; i++) {
+    const point0 = originalIndices[i];
+    const point1 = originalIndices[i + 1];
+    const point2 = originalIndices[i + 2];
+
+    wireframeIndices[index++] = point1;
+    wireframeIndices[index++] = point2;
+    wireframeIndices[index++] = point2;
+    wireframeIndices[index++] = point0;
+  }
+
+  return wireframeIndices;
+}
+
+function createWireframeFromTriangleFan(vertexCount) {
+  const numberOfTriangles = vertexCount - 2;
+  const wireframeIndicesCount = 2 + numberOfTriangles * 4;
+  const wireframeIndices = IndexDatatype.createTypedArray(
+    vertexCount,
+    wireframeIndicesCount
+  );
+  let index = 0;
+
+  // Handle the first edge
+  wireframeIndices[index++] = 0;
+  wireframeIndices[index++] = 1;
+
+  // Add two edges for every triangle in the fan
+  for (let i = 0; i < numberOfTriangles; i++) {
+    wireframeIndices[index++] = i + 1;
+    wireframeIndices[index++] = i + 2;
+    wireframeIndices[index++] = i + 2;
+    wireframeIndices[index++] = 0;
+  }
+
+  return wireframeIndices;
+}
+
+function createWireframeFromTriangleFanIndices(vertexCount, originalIndices) {
+  const originalIndicesCount = originalIndices.length;
+  const numberOfTriangles = originalIndicesCount - 2;
+  const wireframeIndicesCount = 2 + numberOfTriangles * 4;
+  const wireframeIndices = IndexDatatype.createTypedArray(
+    vertexCount,
+    wireframeIndicesCount
+  );
+  let index = 0;
+
+  // Handle the first edge
+  const firstPoint = originalIndices[0];
+  wireframeIndices[index++] = firstPoint;
+  wireframeIndices[index++] = originalIndices[1];
+
+  // Add two edges for every triangle in the fan
+  for (let i = 0; i < numberOfTriangles; i++) {
+    const point1 = originalIndices[i + 1];
+    const point2 = originalIndices[i + 2];
+
+    wireframeIndices[index++] = point1;
+    wireframeIndices[index++] = point2;
+    wireframeIndices[index++] = point2;
+    wireframeIndices[index++] = firstPoint;
+  }
+
+  return wireframeIndices;
+}
+
+/**
+ * Generates a wireframe index buffer for a primitive.
```

Comment:
I would also add that this function both re-indexes existing indices, or it creates them from scratch if `originalIndices` is undefined.